src/ripple/protocol/impl/SecretKey.cpp

```diff
@@ -263,14 +263,15 @@ sign(PublicKey const& pk, SecretKey const& sk, Slice const& m)
                     nullptr) != 1)
                 LogicError("sign: secp256k1_ecdsa_sign failed");
 
-            unsigned char sig[72];
-            size_t len = sizeof(sig);
+            Buffer b(72);
```

Comment:
I feel the existing data structure serves the purpose in this case. Is it necessary to use Buffer here?

It's easier to read and understand `unsigned char sig[72]` imho

Reply:
In _theory_, this should allow the compiler to avoid a copy (see details about [copy elision](https://en.cppreference.com/w/cpp/language/copy_elision#Non-mandatory_copy.2Fmove_.28since_C.2B.2B11.29_elision) and NRVO).

In practice, even if you end up copying, it's just a handful of bytes, and the performance impact (if there is any) will not really be measurable.

_However_, you may be right that this particular change ends up being less clear (and requires the addition of the `resize` member function in `Buffer`). I'll consider reverting this change, thanks!

Reply:
okay, in some ways, this optimization is similar to the placement-new operation. thanks for resource link!